server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -133,6 +134,130 @@ def validate_cdn
       end
     end
 
+    def compatible_cpu_families
+      # @deployment is provided by pre-filter run of find_deployment
+      rhv_hypervisors_list = @deployment.discovered_hosts
+
+      # find the fact_name_id corresponding to "processors"
+      processors_id = FactName.where(name: 'processors').first.id
+
+      # build string containing all cpu families attached to hypervisors
+      processor_concat = ''
+
+      rhv_hypervisors_list.each do |hypervisor|
+        processor_str = hypervisor.fact_values.where("fact_name_id" => processors_id).first.value
+        processor_hash = eval(processor_str)
+        processor_models = processor_hash["models"]
+        processor_concat << (processor_models.join(' ; ').downcase)
+      end
```

Comment:
I keep thinking we should be able to get the FactValues.where( the host id is IN a list of hypervisor ids, and has the processor id found at the top) instead of making N queries we could make one. I'll think about this a bit more and if I find a cleaner way I'll post it here.